filecoin/retriever.go

```diff
@@ -423,10 +450,11 @@ func (retriever *Retriever) queryCandidates(ctx context.Context, retrievalId uui
 	wg.Add(len(candidates))
 
 	for i, candidate := range candidates {
+		candidate := candidate
```

Comment:
without this, `candidate` is reused in the loop and the goroutine ends up using whichever one happens to be set at the time - very likely all goroutines are using the _last_ `candidate` and doing multiple queries!

Reply:
ouch.

Reply:
but thank you for catching!

Reply:
wait, this isn't a bug -- the function takes a candidate parameter and that creates a seperate closure I believe. Was this really failing?

Reply:
I'm not convinced. the function call takes parameters, which should capture seperate varaibles inside the go routine. did you observe this in practice?

Reply:
hmmmm, I'm looking more closely at this today and you're probably right that this isn't what I thought it was (tbh I was in a bit of a hurry and didn't notice it was going in as a function parameter so should be captured). But what I observed was my flaky test becoming apparently non-flaky when I set this and it's flaky exactly in this position - i.e. `RetrievalQueryToPeer` not being called within that goroutine as expected. But I think now this is maybe not the cause, it's something else. I can still observe it flaky without the capture so something else weird is up.

Reply:
ok, found the flaky, my test synchronization wasn't working